### ISSUE TITLE
RecursionError: maximum recursion depth exceeded

### ISSUE THREAD

**Issue by kcem-flyr** [tconbeer/sqlfmt#343]
**Describe the bug**
(...)
```
  File "/lib/python3.10/site-packages/sqlfmt/splitter.py", line 35, in maybe_split
    yield from self.split_at_index(line, i)
  File "/lib/python3.10/site-packages/sqlfmt/splitter.py", line 118, in split_at_index
    self.node_manager.append_newline(head_line)
  File "/lib/python3.10/site-packages/sqlfmt/node_manager.py", line 312, in append_newline
    node = self.create_node(token=nl, previous_node=previous_node)
  File "/lib/python3.10/site-packages/sqlfmt/node_manager.py", line 27, in create_node
    formatting_disabled = self.disable_formatting(token, previous_node)
  File "/lib/python3.10/site-packages/sqlfmt/node_manager.py", line 254, in disable_formatting
    previous_node.formatting_disabled.copy() if previous_node else []
  File "/lib/python3.10/site-packages/sqlfmt/node.py", line 88, in __len__
    return len(str(self))
RecursionError: maximum recursion depth exceeded
```

**To Reproduce**
Query with long enough oneline thing.
[example.txt](https://github.com/tconbeer/sqlfmt/files/10295155/example.txt)

File attached.

**Expected behavior**
Query has been formatted.

**Actual behavior**
```
  File "/lib/python3.10/site-packages/sqlfmt/splitter.py", line 35, in maybe_split
    yield from self.split_at_index(line, i)
  File "/lib/python3.10/site-packages/sqlfmt/splitter.py", line 118, in split_at_index
    self.node_manager.append_newline(head_line)
  File "/lib/python3.10/site-packages/sqlfmt/node_manager.py", line 312, in append_newline
    node = self.create_node(token=nl, previous_node=previous_node)
  File "/lib/python3.10/site-packages/sqlfmt/node_manager.py", line 27, in create_node
    formatting_disabled = self.disable_formatting(token, previous_node)
  File "/lib/python3.10/site-packages/sqlfmt/node_manager.py", line 254, in disable_formatting
    previous_node.formatting_disabled.copy() if previous_node else []
  File "/lib/python3.10/site-packages/sqlfmt/node.py", line 88, in __len__
    return len(str(self))
RecursionError: maximum recursion depth exceeded
```

**Additional context**
What is the output of `sqlfmt --version`?
`sqlfmt, version 0.14.2`

**Comment by tconbeer** [comment]
Oh that's super interesting. Thanks for the report. This should be an easy fix

**Comment by tconbeer** [comment]
@kcem-flyr This patch has been released as v0.14.3, and will be up on PyPI shortly.
